controlplane/kubeadm/internal/machinefilters/machine_filters.go

```diff
@@ -249,7 +246,7 @@ func MatchesKubernetesVersion(kubernetesVersion string) Func {
 }
 
 // MatchesKubeadmBootstrapConfig checks if machine's KubeadmConfigSpec is equivalent with KCP's KubeadmConfigSpec.
-func MatchesKubeadmBootstrapConfig(ctx context.Context, c client.Client, kcp *controlplanev1.KubeadmControlPlane) Func {
+func MatchesKubeadmBootstrapConfig(machineConfigs map[string]*bootstrapv1.KubeadmConfig, kcp *controlplanev1.KubeadmControlPlane) Func {
```

Comment:
It feels odd to pass in a map of KubeadmConfigs here, since it generates a function that takes a single Machine.